### DISCUSSION TITLE
Multiple models for one column in case of multiple foreign keys

### DISCUSSION THREAD

**Discussion by manuelpoelzl** [prisma/prisma#17280] [Q&A] [answered]
Hi,
I wonder if it is somehow possible, or will be possible in the future, to define multiple models for columns?
It is possible in MySQL that a column has several foreign keys on different tables, let's assume I have the following scheme:

```prisma
model EntityReference {
  ID             Int              @id @unique
  Value          String           @unique @db.VarChar(100)
  SharedEntities SharedEntities[]
}

model SharedEntities {
  ID           Int             @id @unique @default(autoincrement())
  EntityType   EntityReference @relation(fields: [EntityTypeId], references: [ID], onDelete: Cascade)
  EntityTypeId Int             @map("EntityType")
  Entity ModelA|ModelB 
}
```
The Entity column under "SharedEntities" in my DB has two foreign keys, one of them goes to the `Entity1` table and the other to the `Entity2` table, both referencing the respective primary key of the table.

Accordingly, the referenced entity can now be either `Entity1` or `Entity2`, but in Prisma only one model can be specified, is there any way I can accomplish this with Prisma?

Another possibility would be to create a separate column under "SharedEntites" for each available entity table and to fill only this column, but this would lead to too many columns for many entities.

Thanks in advance!

**Comment by nurul3101** [comment, marked as answer]
Hello @manuelpoelzl 👋 

At the moment, it's not possible. Prisma needs to know in advance the table referenced by the foreign key. 
We have a related GitHub Issue: [#8976](https://github.com/prisma/prisma/issues/8976)

The only workaround is to have multiple columns for each referencing table.

**Reply by manuelpoelzl** [reply]
All right, thanks for the quick reply!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
